homeassistant/components/zabbix/config_flow.py

```diff
@@ -0,0 +1,706 @@
+        result = await hass.async_add_executor_job(
+            lambda: zabbix_api.itemprototype.create(
+                ruleid=ruleid,
+                hostid=templateid,
+                name="{#KEY}",
+                key="homeassistant.float[{#KEY}]",
+                type=2,
+                value_type=0,
+                delay=0,
+                history="1095d",
+                trends=0,
+                preprocessing=[{"type": 20, "params": "14400"}],
+            ),
+        )
+
+    return templateid
+
+
+async def create_hostname(
+    zabbix_api: ZabbixAPI,
+    hass: HomeAssistant,
+    user_input: dict[str, Any],
+    templateid: str,
+) -> str:
+    """Try to create in Zabbix a hostname to be used to receive entities data with the linked template."""
+
+    result: Any | None
+    groupid: str
+
+    # Check if template already exists
+    try:
+        result = await hass.async_add_executor_job(
+            lambda: zabbix_api.hostgroup.get(
+                output=["groupid", "name"],
+                filter={"name": DEFAULT_ZABBIX_HOSTGROUP_NAME},
+            )
+        )
+        assert isinstance(result, list)
+        groupid = result[0]["groupid"]
+    except (IndexError, KeyError):
+        # No such SHost group - create.
+        result = await hass.async_add_executor_job(
+            lambda: zabbix_api.hostgroup.create(name=DEFAULT_ZABBIX_HOSTGROUP_NAME),
+        )
+        assert isinstance(result, dict)
+        groupid = result["groupids"][0]
+
+    result = await hass.async_add_executor_job(
+        lambda: zabbix_api.host.create(
+            host=user_input[CONF_PUBLISH_STATES_HOST],
+            name=user_input[CONF_PUBLISH_STATES_HOST],
+            groups={"groupid": groupid},
+            templates={"templateid": templateid},
+        ),
+    )
+    assert isinstance(result, dict)
+    return result["hostids"][0]
+
+
+async def zabbix_api_login(hass: HomeAssistant, data: dict[str, Any]) -> ZabbixAPI:
+    """Handle login to ZabbixAPI on each Config flow step."""
+
+    zabbix_api: ZabbixAPI
+
+    zabbix_api = await hass.async_add_executor_job(
+        lambda: ZabbixAPI(url=data.get(CONF_URL))
+    )
+    if data.get(CONF_USE_TOKEN):
+        await hass.async_add_executor_job(
+            lambda: zabbix_api.login(token=data.get(CONF_TOKEN))
+        )
+    else:
+        await hass.async_add_executor_job(
+            lambda: zabbix_api.login(
+                user=data.get(CONF_USERNAME),
+                password=data.get(CONF_PASSWORD),
+            )
+        )
+    await hass.async_add_executor_job(zabbix_api.check_auth)
+
+    return zabbix_api
+
+
+async def zabbix_api_logout(
+    zabbix_api: ZabbixAPI, hass: HomeAssistant, data: dict[str, Any]
+) -> None:
+    """Handle logout from ZabbixAPI if user/password was used to login."""
+
+    if not data.get(CONF_USE_TOKEN):
+        await hass.async_add_executor_job(zabbix_api.logout)
+
+
+class ZabbixConfigFlow(ConfigFlow, domain=DOMAIN):
```

Comment:
I don't know what zabbix is except for that its a monitoring tool. What are we doing here exactly? What data do we need?

Reply:
Right. In this integration Zabbix (monitoring) is used for: 1) sending filtered entities states to zabbix (HA->Zabbix), so in Zabbix we can see same entities as in HA; 2) getting Zabbix filtered problems (triggers) info in HA as sensors (Zabbix->HA). 
Thus we are gathering data required for both scenarios above: host, token or username/password for auth, enities filters, and information for sensors like zabbix hostnames which problems we are interested in.

Reply:
Last week #119784 got opened, maybe you and @rexdek can take a look together if this integration covers both use cases and maybe it will improve this proposal

Reply:
@Wonderer643 
I opened #119784 last week. It is an integration I've been using internally for more than a year now because the current Zabbix integration did not suit my needs. And as a friend of mine wants to use it too, I decided to open a PR. 

My integration imports Zabbix problems (trigger events) and Zabbix services (aggregation of Zabbix problems to report more complex problem states) as sensors into Home Assistant. Further it filters the problems and services to be imported into Home Assistant by the tag / value pairs they have been linked to in Zabbix. This is very different from the features of this Zabbix integration which focusses on exporting Home Assistant states to Zabbix. The only thing that this Zabbix integration imports as a sensor into Home Assistant are problem counts by host. This is something I can easily add to my integration if somebody needs it.

For now I think it should be OK to have two Zabbix integrations in Home Assistant as they serve different purposes. Perhaps we should change the integrations in a way that yours is only concerned with exporting Home Assistant states to Zabbix while mine imports Zabbix states and services into Home Assistant. As said, I can easily add the problem count host sensors to my integration.

Regarding the Zabbix module from PyPI: when developing my integration I first used py-zabbix, too, but switched to pyzabbix (yes, same name but without the dash) later. pyzabbix is maintained, works with authentication tokens and also with the new Zabbix 7.0 version. I was not aware that Zabbix released an official Python library in Feb 24. So I will have a look at it and will rewrite my integration to use it in the next version of my integration.

Reply:
As I see https://github.com/home-assistant/core/pull/119784 uses non-official ZabbixAPI library https://pypi.org/project/pyzabbix/, instead of the https://pypi.org/project/zabbix-utils/. Which opens a threat of future support, as same happened with the currently used in zabbix integration https://pypi.org/project/py-zabbix/ and why bug https://github.com/home-assistant/core/issues/86879 was opened actually.

My main intention for this PR was to solve the bug, move to official python zabbix library and introduce an easy way to configure integration via config_flow for new comers or for the migrated users. 
I was also going to suggest options_flow as next PR to give a way to change some of the filters/sensors configurations.
Now in my sensor realization it is possible to use either token or user/pass to login to Zabbix API and then select one or many hosts from the drop down list which problems user is interested in. But you need to select them one-by-one.
And this is the closest to what is currently in HA for Zabbix integration.

@rexdek sensor realization is a bit different from what is implemented in my solution, it uses only token to login, and it brings some new functionality with using the Zabbix "services" and "tags", which eliminates the need to select hosts one-by-one. But it is not close to the currently available in HA Zabbix integration.

However, I suggest to use my PR for Zabbix integration in order not to break what is already in HA, and then add https://github.com/home-assistant/core/pull/119784 on top of it. If @rexdek agree with this.

Reply:
Hi @joostlek . Any update on the above?
I am resolving this conversation in order to give a PR way to be approved.